app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,47 @@
+class CommentsController < ApplicationController
+
+  def index
+    @comments = Comment.all
+    @post = Post.find(params[:post_id])
+  end
+
+  def new
+      @post = Post.find(params[:post_id])
+      @comment = Comment.new
+  end
+
+  def create
+      @post = Post.find(params[:post_id])
+      @comment = @post.comments.create(comment_params)
+      redirect_to @post
+  end
+
+  def show
+    @post = Post.find(parmas[:id])
+    @comment = Comment.find(params[:id])
+  end
+
+  def edit
+    @post = Post.find(params[:post_id])
+    @comment = Comment.find(params[:id])
+  end
+
+  def update
+    @post = Post.find(params[:post_id])
+    @comment = Comment.find(params[:id])
```

Comment:
Recommend `@post.comments.find...`  This limits it to a comment for this post (even if someone hijacks the request.